### ISSUE TITLE
unnecessary_lazy_evaluations could give more targeted suggestion when unwrapping

### ISSUE THREAD

**Issue by illicitonion** [rust-lang/rust-clippy#15465]
### Summary

Given this code which ensures `definitely` contains a value:

```rust
let maybe: Option<u32> = None;
let definitely = maybe.or_else(|| Some(7)).unwrap();
println!("{definitely}");
```

Better code would be:

```rust
let definitely = maybe.unwrap_or(7);
```

Right now clippy suggests `maybe.or(Some(7)).unwrap()`:

```
warning: unnecessary closure used to substitute value for `Option::None`
 --> src/main.rs:3:22
  |
3 |     let definitely = maybe.or_else(|| Some(7)).unwrap();
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations
  = note: `#[warn(clippy::unnecessary_lazy_evaluations)]` on by default
help: use `or` instead
  |
3 -     let definitely = maybe.or_else(|| Some(7)).unwrap();
3 +     let definitely = maybe.or(Some(7)).unwrap();
  |
```

It'd be great to notice the `unwrap` and have the suggestion be `unwrap_or` (or `unwrap_or_else`) rather than `or(_).unwrap()`

### Reproducer

I tried this code:

```rust
fn main() {
    let maybe: Option<u32> = None;
    let definitely = maybe.or_else(|| Some(7)).unwrap();
    println!("{definitely}");
}
```

I expected to see this happen: Suggestion to use `unwrap_or`

Instead, this happened: Suggestion to use `or` and then `unwrap`

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: aarch64-apple-darwin
release: 1.88.0
LLVM version: 20.1.5
```

**Comment by ada4a** [comment]
not really a bug..

@rustbot label -C-bug +C-enhancement

**Comment by ada4a** [comment]
In fact, there is a lint that suggests exactly that second transformation: [`or_then_unwrap`](https://rust-lang.github.io/rust-clippy/master/index.html#or_then_unwrap). So if you run `cargo clippy --fix`, it will first apply the suggestion of `unnecessary_lazy_evaluations`, and then that of `or_then_unwrap`. Such composition is expected in Clippy, because it allows each individual lint to concentrate on what _it_ needs to fix, instead of thinking of all the ways the fixed code could theoretically be improved further.

@rustbot label S-triage
